iceoryx_hoofs/vocabulary/include/iox/detail/string_internal.hpp

```diff
@@ -111,10 +110,16 @@ struct GetSize<char>
     }
 };
 
-/// @brief struct to get a pointer to the char array of the fixed string/string literal/std::string
+/// @brief generic empty implementation of the struct to get the data of a string
 template <typename T>
-struct GetData;
+struct GetData
```

Comment:
I'm not sure if that is a good idea. You mean to adding a `static_assert` to all of the types which inherit from `std::false_type`? E.g. in `iox/type_traits.hpp`. That would lead to the fact that "Substitution Failure Is Not An Error" (SFINAE) would become "Substitution Failure Is An Error", right? And thereby defeating the purpose.